script.js

```diff
@@ -0,0 +1,217 @@
+    if (americanFloorNumber <= 12) {
+        return americanFloorNumber - 1;
+    }
+    if (americanFloorNumber >= 14) {
+        return americanFloorNumber - 2;
+    }
+}
+
+console.log(getTheFloorInTheEuropeanSystem(14));
+
+// 7. In the rock-paper-scissors.js file.
+
+// 8. Write the isValidPassword function.
+// Make sure that the password is at least 8 characters long and contains at least one number and one uppercase letter.
+
+function isLettersNumber(word) {
+    return word.length >= 8;
+
+}
+
+function isUppercaseLetter(word) {
+    const uppercaseLetters =  /[A-Z]/g;
+    const matchingResult = word.match(uppercaseLetters);
+    return matchingResult !== null;
+
+}
+
+function isNumber(word) {
+    const numbers = /\d/g;
+    const matchingResult = word.match(numbers);
+    return matchingResult !== null;
+}
+
+function isValidPassword(password) {
+    return isLettersNumber(password) &&
+        isUppercaseLetter(password) &&
+        isNumber(password);
+}
+
+console.log(isValidPassword('Password123'));
+
+// 9. In the getPasswordStrength.js file.
+
+// 10. Create the getTotalOrderCost function that returns the total cost of the order and accepts the following arguments:
+// - the base cost of the items
+// - whether the customer is a VIP member - they get a 5% discount
+// - how many loyalty points the customer wants to use - each one of them reduces the cost by a cent (the final cost should not be lower than $0)
+// - is the package going to be shipped internationally - shipping internationally costs $10 (otherwise, it costs $5)
+// Make sure to apply the VIP discount before the loyalty points. The VIP discount does not apply to shipping.
+
+function getTheCostForVIPMember(cost) {
+    return cost * 0.05;
+}
+
+function getLoyaltyPointsDiscount(points) {
+    return  points / 100;
+}
+
+function getShippedInternationally(shippingType) {
+    if (shippingType === 'international') {
+        return 10;
+    }
+    return 5;
+}
+
+function getTotalOrderCost(baseCost, isVipMember, loyaltyPoints, shippingType) {
+    const shippingCost = getShippedInternationally(shippingType);
+    if (isVipMember) {
+        return baseCost - getTheCostForVIPMember(baseCost) - getLoyaltyPointsDiscount(loyaltyPoints) + shippingCost;
+    }
+    return baseCost - getLoyaltyPointsDiscount(loyaltyPoints) + shippingCost;
+}
+
+console.log(getTotalOrderCost(100, '', 5,  'international'));
+
+// 11. Create the getTicketPrice function that returns the ticket price and accepts the following arguments:
+// - the base price of the ticket
+// - how many days are there until the show - buying tickets more than a month in advance provides a 10% discount
+// - is the show on the weekend - weekend shows are $15 more expensive
+// The discount applied if the tickets are bought a month in advance should decrease the $15 charged if the show plays on the weekend.
+
+function getDiscountByNumberOfDays(daysNumber) {
+    if (daysNumber > 30) {
+        return 1 - 0.10;
```

Comment:
This is not correct. If you multiply the price by 0.9, you decrease it instead of increasing it.